addons/l10n_es_edi_tbai/models/account_edi_format.py

```diff
@@ -0,0 +1,573 @@
+        xsd_name = get_key(tax_agency, 'xsd_name')['cancel' if cancel else 'post']
+
+        try:
+            validate_xml_from_attachment(self.env, xml_doc, xsd_name, prefix='l10n_es_edi_tbai')
+        except UserError as e:
+            return {'error': str(e), 'blocking_level': 'error'}
+        return {}
+
+    def _l10n_es_tbai_get_invoice_content_edi(self, invoice):
+        cancel = invoice.edi_state in ('to_cancel', 'cancelled')
+        xml_tree = self._get_l10n_es_tbai_invoice_xml(invoice, cancel)[invoice]['xml_file']
+        return etree.tostring(xml_tree)
+
+    def _get_l10n_es_tbai_invoice_xml(self, invoice, cancel=False):
+        # If previously generated XML was posted and not rejected (success or timeout), reuse it
+        doc = invoice._get_l10n_es_tbai_submitted_xml(cancel)
+        if doc is not None:
+            return {invoice: {'xml_file': doc}}
+
+        # Otherwise, generate a new XML
+        values = {
+            **invoice.company_id._get_l10n_es_tbai_license_dict(),
+            'is_emission': not cancel,
+            'datetime_now': datetime.now(tz=timezone('Europe/Madrid')),
+            'format_date': lambda d: datetime.strftime(d, '%d-%m-%Y'),
+            'format_time': lambda d: datetime.strftime(d, '%H:%M:%S'),
+            'format_float': lambda f: float_repr(f, precision_digits=2),
+        }
+        values.update(self._l10n_es_tbai_get_header_values(invoice))
+        values.update(self._l10n_es_tbai_get_subject_values(invoice, cancel))
+        values.update(self._l10n_es_tbai_get_invoice_values(invoice, cancel))
+        values.update(self._l10n_es_tbai_get_trail_values(invoice, cancel))
+        template_name = 'l10n_es_edi_tbai.template_invoice_main' + ('_cancel' if cancel else '_post')
+        xml_str = self.env['ir.qweb']._render(template_name, values)
+        xml_doc = cleanup_xml_node(xml_str)
+        xml_doc = self._l10n_es_tbai_sign_invoice(invoice, xml_doc)
+        res = {invoice: {'xml_file': xml_doc}}
+
+        # Optional check using the XSD
+        res[invoice].update(self._l10n_es_tbai_validate_xml_with_xsd(xml_doc, cancel, invoice.company_id.l10n_es_tbai_tax_agency))
+        return res
+
+    def _l10n_es_tbai_get_header_values(self, invoice):
+        return {
+            'tbai_version': self.L10N_ES_TBAI_VERSION,
+            'odoo_version': release.version,
+        }
+
+    def _l10n_es_tbai_get_subject_values(self, invoice, cancel):
+        # === SENDER (EMISOR) ===
+        sender = invoice.company_id
+        values = {
+            'sender_vat': sender.vat[2:] if sender.vat.startswith('ES') else sender.vat,
+            'sender': sender,
+        }
+        if cancel:
+            return values  # cancellation invoices do not specify recipients (they stay the same)
+
+        # NOTE: TicketBai supports simplified invoices WITH recipients: use checkbox in invoice (optional until POS implemented)
```

Comment:
Do we have that checkbox or just the simplified partner?  Right now, simplified invoices are only sent if the partner is the simplified partner (like the consumidor final)  Is that still what we want?  @chklop

Reply:
We don't have that checkbox, I modified the comment to reflect that.

Reply:
That's something we need to follow-up and crack with the PoS afterwards for Spain. The contacts from the Spanish office indeed ask us to look at sending simplified invoices for the PoS, and it's actually more complicated that that because you can't create simplified invoices for amounts greater than a specific amount (somewhere around 400€ I think, need to double check). 

Can we skip that for now and tackle that later in a l10n_pos_es task though ?

Reply:
That was the intention AFAIK.